src/main/java/com/elytradev/architecture/compat/ArchitectConverter.java

```diff
@@ -0,0 +1,276 @@
+package com.elytradev.architecture.compat;
+
+import com.elytradev.architecture.common.ArchitectureMod;
+import com.elytradev.architecture.common.shape.Shape;
+import com.elytradev.architecture.common.shape.ShapeKind;
+import com.elytradev.architecture.common.tile.TileShape;
+import li.cil.architect.api.ConverterAPI;
+import li.cil.architect.api.converter.Converter;
+import li.cil.architect.api.converter.MaterialSource;
+import li.cil.architect.api.converter.SortIndex;
+import net.minecraft.block.Block;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTBase;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.util.Rotation;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+import net.minecraftforge.fluids.FluidStack;
+import net.minecraftforge.items.IItemHandler;
+import scala.actors.threadpool.Arrays;
+
+import java.util.Collections;
+import java.util.UUID;
+import java.util.function.Predicate;
+
+/**
+ * Architect integration. Ensures that the right item is pulled from the player's inventory and that
+ * rotations are applied correctly.
+ *
+ * @author Stan Hebben
+ */
+public class ArchitectConverter implements Converter {
+    private static final ArchitectConverter INSTANCE = new ArchitectConverter();
+
+    // never change this (if you do, it will break existing Architect blueprints)
+    private static final UUID CONVERTER_UUID = UUID.fromString("182148b6-fba8-4acb-95a5-66409d34eb59");
```

Comment:
Wait what? Does the architect api make you use UUIDs instead of ResourceLocations???

Reply:
Yes, the Architect API uses UUIDs to identify its converters.